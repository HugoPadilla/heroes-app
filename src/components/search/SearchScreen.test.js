const { mount } = require("enzyme");
const { MemoryRouter } = require("react-router-dom");
const { SearchScreen } = require("./SearchScreen");

const mockNavigate = jest.fn();

jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  useNavigate: () => mockNavigate,
}));

describe("Pruebas en <SearchScreen />", () => {
  test("debe de mostrarse corectamente con valores por defectos", () => {
    const wrapper = mount(
      <MemoryRouter initialEntries={["/search"]}>
        <SearchScreen />
      </MemoryRouter>
    );
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find(".alert-info").text().trim()).toBe("Busca un heroes");
  });

  test("debe de mostrar a Batman y el input con el valor del queryString", () => {
    const wrapper = mount(
      <MemoryRouter initialEntries={["/search?q=batman"]}>
        <SearchScreen />
      </MemoryRouter>
    );

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("input").prop("value")).toBe("batman");
  });

  test("debe de mostrar un error si no se encuentra el hero", () => {
    const wrapper = mount(
      <MemoryRouter initialEntries={["/search?q=heroe"]}>
        <SearchScreen />
      </MemoryRouter>
    );

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find(".alert-danger").text().trim()).toBe(
      "No hay resultados para: heroe"
    );
  });

  test("debe de llamar al metodo handleSearch con el valor del input", () => {
    const wrapper = mount(
      <MemoryRouter initialEntries={["/search?q=batman"]}>
        <SearchScreen />
      </MemoryRouter>
    );

    wrapper.find("input").simulate("submit", {
      target: {
        name: "searchText",
        value: "batman",
      },
    });

    wrapper.find("form").prop("onSubmit")({
      preventDefault: () => {},
    });

    expect(mockNavigate).toHaveBeenCalledWith("?q=batman");
  });
});
