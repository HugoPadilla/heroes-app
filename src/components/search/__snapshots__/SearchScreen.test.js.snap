// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <SearchScreen /> debe de mostrar a Batman y el input con el valor del queryString 1`] = `
Array [
  <h1>
    Busquedas
  </h1>,
  <hr />,
  <div
    className="row"
  >
    <div
      className="col-5"
    >
      <h4>
        Buscar
      </h4>
      <hr />
      <form
        onSubmit={[Function]}
      >
        <input
          autoComplete="off"
          className="form-control"
          name="searchText"
          onChange={[Function]}
          placeholder="Buscar un heroe"
          type="text"
          value="batman"
        />
        <button
          className="btn btn-outline-primary mt-1"
          type="submit"
        >
          Buscar...
        </button>
      </form>
    </div>
    <div
      className="col-7"
    >
      <h4>
        Resultado
      </h4>
      <hr />
      <div
        className="col"
      >
        <div
          className="card"
        >
          <img
            alt="Batman"
            className="card-img-top"
            src="/assets/dc-batman.jpg"
          />
          <div
            className="card-body"
          >
            <h5
              className="card-title"
            >
              Batman
            </h5>
            <p
              className="card-text"
            >
              Bruce Wayne
            </p>
            <p
              className="card-text"
            >
              <small
                className="text-muted"
              >
                Detective Comics #27
              </small>
            </p>
            <a
              className="btn btn-outline-primary"
              href="/hero/dc-batman"
              onClick={[Function]}
            >
              Ver mas...
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>,
]
`;

exports[`Pruebas en <SearchScreen /> debe de mostrar un error si no se encuentra el hero 1`] = `
Array [
  <h1>
    Busquedas
  </h1>,
  <hr />,
  <div
    className="row"
  >
    <div
      className="col-5"
    >
      <h4>
        Buscar
      </h4>
      <hr />
      <form
        onSubmit={[Function]}
      >
        <input
          autoComplete="off"
          className="form-control"
          name="searchText"
          onChange={[Function]}
          placeholder="Buscar un heroe"
          type="text"
          value="heroe"
        />
        <button
          className="btn btn-outline-primary mt-1"
          type="submit"
        >
          Buscar...
        </button>
      </form>
    </div>
    <div
      className="col-7"
    >
      <h4>
        Resultado
      </h4>
      <hr />
      <div
        className="alert alert-danger"
      >
        No hay resultados para: 
        heroe
      </div>
    </div>
  </div>,
]
`;

exports[`Pruebas en <SearchScreen /> debe de mostrarse corectamente con valores por defectos 1`] = `
Array [
  <h1>
    Busquedas
  </h1>,
  <hr />,
  <div
    className="row"
  >
    <div
      className="col-5"
    >
      <h4>
        Buscar
      </h4>
      <hr />
      <form
        onSubmit={[Function]}
      >
        <input
          autoComplete="off"
          className="form-control"
          name="searchText"
          onChange={[Function]}
          placeholder="Buscar un heroe"
          type="text"
          value=""
        />
        <button
          className="btn btn-outline-primary mt-1"
          type="submit"
        >
          Buscar...
        </button>
      </form>
    </div>
    <div
      className="col-7"
    >
      <h4>
        Resultado
      </h4>
      <hr />
      <div
        className="alert alert-info"
      >
        Busca un heroes
      </div>
    </div>
  </div>,
]
`;
